packages/components/stepper/src/Stepper.tsx

```diff
@@ -1,7 +1,144 @@
-import { ComponentPropsWithoutRef, forwardRef, PropsWithChildren } from 'react'
+import { useFormFieldControl } from '@spark-ui/form-field'
+import { Icon } from '@spark-ui/icon'
+import { Minus } from '@spark-ui/icons/dist/icons/Minus'
+import { Plus } from '@spark-ui/icons/dist/icons/Plus'
+import { InputGroup } from '@spark-ui/input'
+import {
+  Children,
+  cloneElement,
+  type FC,
+  forwardRef,
+  isValidElement,
+  type PropsWithChildren,
+  type ReactElement,
+  useCallback,
+  useRef,
+} from 'react'
 
-export type StepperProps = ComponentPropsWithoutRef<'div'>
+import { StepperDecrementButton, StepperIncrementButton } from './StepperButton'
+import { StepperInput } from './StepperInput'
+import type { StepperButtonProps, StepperProps } from './types'
+import { useStepper } from './useStepper'
 
-export const Stepper = forwardRef<HTMLDivElement, PropsWithChildren<StepperProps>>((props, ref) => {
-  return <div ref={ref} {...props} />
-})
+export const Stepper = forwardRef<HTMLDivElement, PropsWithChildren<StepperProps>>(
+  (
+    {
+      children,
+      formatOptions,
+      minValue,
+      maxValue,
+      incrementAriaLabel,
+      decrementAriaLabel,
+      ...stepperProps
+    },
+    forwardedRef
+  ) => {
+    const inputRef = useRef(null)
+
+    const {
+      groupProps,
+      inputProps: _inputProps,
+      incrementButtonProps: _incrementButtonProps,
+      decrementButtonProps: _decrementButtonProps,
+    } = useStepper({
+      ...stepperProps,
+      formatOptions,
+      minValue,
+      maxValue,
+      incrementAriaLabel,
+      decrementAriaLabel,
+      inputRef,
+    })
+
+    const findElement = useCallback(
+      (elementDisplayName: string): ReactElement | undefined => {
+        const childrenArray = Children.toArray(children)
+
+        return childrenArray
+          .filter(isValidElement)
+          .find(child =>
+            (child.type as FC & { displayName?: string }).displayName?.includes(elementDisplayName)
+          )
+      },
+      [children]
+    )
+
+    const incrementBtnFromChildren = findElement('Stepper.IncrementButton')
```

Comment:
The `findElement` pattern is a anti-pattern that goes against our composability rules, we decided to stop using it. Is there a reason you must use it ?

Reply:
Let's discuss it when you come back :) I agree it is necessary in some cases (I think only Select and Snackbar it is justified). I think it should be eliminated from Input too, but it is a breaking change and is not an emergency.